### ISSUE TITLE
Kill-All Switch for External Resources

### ISSUE THREAD

**Issue by H0d3nk0b0ld** [Zettlr/Zettlr#4681]
### Checklist

- [X] Zettlr does not possess this feature (if it does, but the feature doesn't work out for you, propose an enhancement instead.)
- [X] Zettlr's current features cannot be used to approximate this behaviour (otherwise, propose an enhancement instead.)
- [X] I have searched the documentation to make sure I did not overlook it.
- [ ] I have searched both open and closed issues to see if someone has already proposed this.

### Feature Proposal

It would be useful to have a Setting that prevents any kind of remote Resource to be loaded.

This would (if activated) prevent IFrames from loading.

Also when Ctrl + Clicking a link a confirmation popup could be shown:
"When you click confirm your default Web Browser will open www.example.com."

Edit: It would also prevent the fetching of Link-Previews and whatever features I am forgetting here :P

### Example Workflow

Users would go to the Settings and disable remote Resources in order to harden their security.

### Caveats

If enabled, this would prevent IFrames and the built-in Browser from working properly.

### Do you wish to attempt implementing this yourself?

No

### Platforms

- [ ] Windows
- [ ] macOS
- [X] Linux

**Comment by nathanlesage** [comment]
This would mean an unreasonable amount of work. I do understand that some people like to make their app entirely offline to ensure that nothing malicious can be loaded. However, as I've outlined in the [security notes](https://docs.zettlr.com/en/getting-started/a-note-on-security/), this will always be incomplete and not the way it's meant to be. Therefore, implementing a form of "kill switch" would be lying to the users, because I cannot entirely kill all network accesses within electron (and even less so the network accesses Pandoc makes); the only way to properly do that is to use an external program that simply prevents any network access by the binary itself. And I know that this is what some people have done.

On my side, I will try to ensure that Zettlr works fine even in these circumstances (i.e. that failing network accesses do not make the UI unresponsive or broken), but I cannot provide such a switch myself; for that the Electron framework is unfortunately too limited. Besides, it is mainly a fringe interest – this is not intended to come across as derogatory, but 99% of Zettlr users wouldn't need or want such a button; and people who do need it are able to accomplish this with an external app.

Bottom line is: Implementing this would mean an unreasonable amount of work that is not even guaranteed to do what it says on the can, so I'm opting not to do this and hope you can understand my reasoning.
